tensorflow/core/util/gpu_device_functions.h

```diff
@@ -878,6 +931,28 @@ __device__ inline std::complex<double> operator/(
 }
 #endif  // GOOGLE_CUDA
 
+namespace functor {
+// ROCm hcc(clang) has severe difficulties dealing with std::complex directly due to a header issue.
+// This template assists in casting std::complex into the corresponding internal ROCm types.
+template <class T> 
+struct CpxMapT {
```

Comment:
`MAP_STD_COMPLEX_TO_HIP_COMPLEX` ?

Reply:
Too long I think?

Reply:
your call, but a more descriptive version than the current incarnation would be better. 
this is admittedly nit-picking :)